### PR TITLE
Refactor automation using terraform

### PR DESCRIPTION
**Pull request by ShylajaDevadiga**
Signed-off-by: Shylaja Devadiga <shylaja.devadiga@suse.com>

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

**Review by dereknola — APPROVED** [review]
Don't forget to add a blurb to the TESTING.md on this section.

File: tests/terraform/Dockerfile.build [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM golang:alpine
+
+ARG TERRAFORM_VERSION=0.12.10
+ENV TERRAFORM_VERSION=$TERRAFORM_VERSION
+
+
+RUN apk update && \
+    apk upgrade --update-cache --available && \
+    apk add curl git jq bash openssh unzip gcc g++ make ca-certificates && \
+    curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (briandowns) [inline tests/terraform/Dockerfile.build:10]: Why are we using dep instead of go mods?

File: tests/terraform/amd64_resource_files/secrets.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: e2e-secret5
+type: Opaque
+stringData:
+  config.yaml: |
+    key: "last"
+    val: "call"
```
Comment (briandowns) [inline tests/terraform/amd64_resource_files/secrets.yaml:49]: Add new line

File: tests/terraform/createcluster.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+func BuildCluster(nodeOs, clusterType, externalDb, resourceName string, t *testing.T, destroy bool) (string, string, string) {
+
+	tDir := "./modules/k3scluster"
+	vDir := "/config/" + nodeOs + clusterType + ".tfvars"
+
+	if externalDb != "" {
+		vDir = "/config/" + nodeOs + externalDb + ".tfvars"
+	}
+
+	tfDir, _ := filepath.Abs(tDir)
```
Comment (briandowns) [inline tests/terraform/createcluster.go:36]: Is it safe to be ignoring the errors here?

File: tests/terraform/createcluster.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	if destroy {
+		fmt.Printf("Cluster is being deleted")
+		terraform.Destroy(t, TerraformOptions)
+		return "", "", ""
+	}
+
+	fmt.Printf("Creating Cluster")
+	terraform.InitAndApply(t, TerraformOptions)
+	kubeconfig := terraform.Output(t, TerraformOptions, "kubeconfig") + "_kubeconfig"
+	masterIps := terraform.Output(t, TerraformOptions, "master_ips")
```
Comment (briandowns) [inline tests/terraform/createcluster.go:57]: Nit:

```suggestion
	masterIPs := terraform.Output(t, TerraformOptions, "master_ips")
```

Same would go for the variable name below.

File: tests/terraform/testutils.go [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	conn, err := ssh.Dial("tcp", host, config)
+	checkError(err)
+	return conn
+}
+
+/*
+func runsshCommand(cmd string, conn *ssh.Client) (string, error) {
```
Comment (briandowns) [inline tests/terraform/testutils.go:72]: Should this be removed?

File: tests/terraform/Dockerfile.build (hunk 1 above)
Comment (ShylajaDevadiga) [reply tests/terraform/Dockerfile.build]: Updated. Thanks

File: tests/terraform/createcluster.go (hunk 4 above)
Comment (ShylajaDevadiga) [reply tests/terraform/createcluster.go]: Updated

File: tests/terraform/testutils.go (hunk 5 above)
Comment (ShylajaDevadiga) [reply tests/terraform/testutils.go:72]: Updated.

**Pull request merged**